### PR TITLE
Gobindata 2.. Its back.  And its hungry.

### PR DESCRIPTION
**Pull request by jayunit100**
This is a gobindata implementation which honors the decision made in @kubernetes/sig-testing yesterday, based on offline idea suggested by @timothysc.  It embodies 3 ideas.
- kill disk file dependencies
- embed all necessary data in gobindata
- stream gobindata contents directly into standard input.

should _just work_ ... testing offline while putting this PR up for review.

### CODE REVIEW

**Comment by jayunit100** [comment]
@fejta i think this guy is ready for you .  IT builds on top #24806 , and i kept the commits separated.  its mostly a minimal change once you distill it down.    

- commit 1: kubectl stdin (already LGTM, pending merge)
- commit 2 : gobindata , first iteration with temporary file
- commit 3 gobindata, final iteration, modified to leverage the stdin stuff, so no need for tmp files.

I will rebase 2+3 once you give it the thumbs up.  I've kept the history in the commits for context , but not because we actually need it.

**Comment by jayunit100** [comment]
ok, pushing final rebase.  above looks ok except for one test, which i missed adding to bindata.
so safe to lose history now and do a final review :)

File: test/e2e/framework/gobindata_util.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+//
+// # Generate the bindata file.
+// go-bindata \
+// 	-pkg generated -ignore .jpg -ignore .png -ignore .md \
+// 	./examples/* ./docs/user-guide/* test/e2e/testing-manifests/kubectl/* test/images/*
+//
+// # Copy it into the generated directory if the results are what you expected.
+// cp bindata.go test/e2e/generated
+func ReadOrDie(filePath string) []byte {
+	fileBytes, err := generated.Asset(filePath)
```
Comment (timothysc) [inline test/e2e/framework/gobindata_util.go]: This is so much better!

**Comment by jayunit100** [comment]
Looks like I forgot to remove reporoot from the updated patch, will patch that tonite.  Otherwise ci seems happy now with the test/images files.

File: hack/verify-flags/known-flags.txt [hunk 2]
```diff
@@ -144,6 +144,7 @@ gather-resource-usage
 gce-project
 gce-service-account
 gce-zone
+generated-assets
```
Comment (fejta) [inline hack/verify-flags/known-flags.txt]: Remove?

Comment (jayunit100) [inline hack/verify-flags/known-flags.txt]: no need for any assets anymore i need to remove this entirely

**Comment by timothysc** [comment]
might want to denote not to merge till the kube stdin stuff is done.

File: test/e2e/framework/test_context.go [hunk 3]
```diff
@@ -26,12 +26,15 @@ import (
 )
 
 type TestContextType struct {
-	KubeConfig            string
-	KubeContext           string
-	KubeVolumeDir         string
-	CertDir               string
-	Host                  string
-	RepoRoot              string
+	KubeConfig    string
```
Comment (fejta) [inline test/e2e/framework/test_context.go]: Can we revert these changes? (aka set the padding back to the previous version?)

**Comment by fejta** [comment]
@jayunit100 do you have the ability to trigger tests? Also does it make sense to squash this all into one commit? Looks good otherwise (I had one comment about indentation)

File: test/e2e/framework/test_context.go (hunk 3 above)
Comment (jayunit100) [inline test/e2e/framework/test_context.go]: I think gofmt did this ? Will double check

**Comment by jayunit100** [comment]
@fejta
- this patch duplicates another kubectl inling patch which is a separate PR, so i kinda think  2 commits is better?  
- removed the changes to RepoRoot comments which forced reformatting.

**Comment by fejta** [comment]
LGTM but agree on getting the other change in first before adding the tag. @jayunit100 @timothysc does this need a release note?

**Comment by jayunit100** [comment]
I thought about this some.   I'll create a separate issue for automating and describe the 3 possible approaches..

**Comment by timothysc** [comment]
I don't think a release note is needed on this PR itself as it's the last hurdle, but I do think we should have a release note that conformance tests are now stand-alone from a binary perspective.

**Comment by jayunit100** [comment]
Here we go rebasing

**Comment by jayunit100** [comment]
@timothysc PTAL needed a quick rebase.... no effective changes other than one line in boilerplate.py

**Comment by fejta** [comment]
Looks like it is failing `[It] should check if kubectl describe prints relevant information for rc and pods [Conformance]`

**Comment by jayunit100** [comment]
hmmmm thats odd, i wonder what changed? will look

**Comment by jayunit100** [comment]
Oh, duh, the examples got updated i think.  anyway, pushed an updated gobindata, should work now

**Comment by timothysc** [comment]
Still waiting on stdin to land.

**Comment by jayunit100** [comment]
closing in favor of kubectlstdin mega-patch
